yb-voyager/cmd/exportData.go

```diff
@@ -127,29 +129,119 @@ func exportDataCommandFn(cmd *cobra.Command, args []string) {
 
 	success := exportData()
 	if success {
-		tableRowCount := getExportedRowCountSnapshot(exportDir)
-		callhome.GetPayload(exportDir, migrationUUID)
-		callhome.UpdateDataStats(exportDir, tableRowCount)
-		callhome.PackAndSendPayload(exportDir)
+		packAndSendExportDataPayload(COMPLETE)
 
 		setDataIsExported()
 		color.Green("Export of data complete")
 		log.Info("Export of data completed.")
 		startFallBackSetupIfRequired()
 	} else if ProcessShutdownRequested {
 		log.Info("Shutting down as SIGINT/SIGTERM received.")
+		packAndSendExportDataPayload(EXIT)
 	} else {
 		color.Red("Export of data failed! Check %s/logs for more details.", exportDir)
 		log.Error("Export of data failed.")
+		packAndSendExportDataPayload(ERROR)
 		atexit.Exit(1)
 	}
 }
 
+func packAndSendExportDataPayload(status string) {
+	//TODO: send this INPROGRESS status in some fixed interval for long running export data
+
+	if !callhome.SendDiagnostics {
```

Comment:
Can you add this logic to callhome.SendPayload function as an extra layer of protection?